editor/test/effects.js

```diff
@@ -319,6 +319,117 @@ describe( 'effects', () => {
 			expect( dispatch ).toHaveBeenCalledTimes( 1 );
 			expect( dispatch ).toHaveBeenCalledWith( requestMetaBoxUpdates( [ 'normal', 'side' ] ) );
 		} );
+
+		it( 'should dispatch notices when reverting a published post to a draft.', () => {
+			const handler = effects.REQUEST_POST_UPDATE_SUCCESS;
+			const dispatch = jest.fn();
+			const store = { getState: () => {}, dispatch };
+
+			selectors.getDirtyMetaBoxes.mockReturnValue( [ 'normal', 'side' ] );
+
+			const post = {
+				id: 1,
+				title: {
+					raw: 'A History of Pork',
+				},
+				content: {
+					raw: '',
+				},
+				status: 'publish',
+			};
+
+			const previousPost = {
+				...post,
+				status: 'draft',
+			};
+
+			handler( { post, previousPost }, store );
+
+			expect( dispatch ).toHaveBeenCalledTimes( 2 );
+			expect( dispatch.mock.calls[ 0 ][ 0 ] ).toEqual( {
```

Comment:
Minor: A bit fragile to assume the order in which the notice dispatch is called.